doc/Programming-guide/Your-first-Kaa-application/attach/demo-cpp/KaaDemo.cpp

```diff
@@ -25,22 +25,57 @@
 #include <vector>
 #include <list>
 
-// using namespace kaa;
-
-class Transceiver : public kaa::IConfigurationReceiver {
- private:
-    // Static variables
-    static constexpr auto savedConfig = "saved_config.cfg";
-    std::shared_ptr<kaa::IKaaClient> kaaClient;
-    int32_t samplePeriod;
-    boost::asio::io_service service;
-    boost::posix_time::seconds interval;
-    boost::asio::deadline_timer timer;
+class ConfigurationCollection : public kaa::IConfigurationReceiver {
+public:
+    ConfigurationCollection()
+        : kaaClient(kaa::Kaa::newClient())
+        , interval(samplePeriod = 0)
+        , timer(service, interval) 
+    {
+        // Set a custom strategy for uploading logs.
+        kaaClient->setLogUploadStrategy(
+            std::make_shared<kaa::RecordCountLogUploadStrategy>
+            (1, kaaClient->getKaaClientContext()));
+        // Set up a configuration subsystem.
+        kaa::IConfigurationStoragePtr storage(
+            std::make_shared<kaa::FileConfigurationStorage> (
+            std::string(savedConfig)));
+        kaaClient->setConfigurationStorage(storage);
+        kaaClient->addConfigurationListener(*this);
+        auto lambda = [this](const boost::system::error_code& err) {
+            this->update();
+        };
+        timer.async_wait(lambda);
+    }
+    
+    void run() 
+    {
+        // Run the Kaa endpoint.
+        kaaClient->start();
+        service.run();
+    }
+    
+    ~Transceiver() 
+    {
+        // Stop the Kaa endpoint.
+        kaaClient->stop();
+        std::cout << "Simple client demo stopped" << std::endl;
+    }
+    
+private:
+    static constexpr auto savedConfig_ = "saved_config.cfg";
+    std::shared_ptr<kaa::IKaaClient> kaaClient_;
+    int32_t samplePeriod_;
```

Comment:
I don't think that compiles. The variable is called `samplePeriod` in other places.